extension.js

```diff
@@ -1,57 +1,160 @@
-const vscode = require('vscode');
+const { Location, Range, Uri, Position, workspace, commands, languages } = require('vscode');
 const path = require('path');
 
 class ReferenceProvider {
 
+    constructor() {
+        this.childWord = '';
+    }
+
+     /**
+     * Remove all white space and new lines from str
+     * @param {String} str
+     * @return {String}
+     */
+    stripWhiteSpaceAndNewLines(str) {
+        return (str + '').replace(/\s|\r|\n/g, '');
+    }
+
     /**
      * Get a require or define statement from string
      * @param {String} str
      * @return {String} 
      */
     getRequireOrDefineStatement(str) {
-        const match = /(define|require)\s*\(([^)]*)/gi.exec(str + "");
+        str = this.stripWhiteSpaceAndNewLines(str);
+        const match = /(define|require).*{/i.exec(str);
         return match && match[0] || null;
     }
 
     /**
+     * Get partial code from require/define statement until given line and character
+     * @param {VSCODE Document} document
+     * @param {Int} line to stop
+     * @param {Int} characterIndex to stop
+     * @return {String} substring for given document ending at given line and character and starting at last define or require
+     */
+    getRequireOrDefineCodeUntilCharacterIndex(document, line, characterIndex) {
+        const textBeforeChar = document.getText(new Range(0, 0, line, characterIndex));
+        const textBeforeCharWithoutComments = this.stripAllComments(textBeforeChar);
+        const lastOccuranceDefine = textBeforeCharWithoutComments.toLowerCase().lastIndexOf('define');
+        const lastOccuranceRequire = textBeforeCharWithoutComments.toLowerCase().lastIndexOf('require');
+        const lastOccuranceRequireOrDefine = lastOccuranceDefine > lastOccuranceRequire ? lastOccuranceDefine : lastOccuranceRequire;
+
+        return textBeforeCharWithoutComments.substr(lastOccuranceRequireOrDefine > -1 ? lastOccuranceRequireOrDefine : 0);
+    }
+
+    /**
+     * Check if string contains multiple define or require statements
+     * @param {String} str to search in 
+     * @return {Bool}
+     */
+    stringHasMultipleDefineOrRequireStatements(str) {
+        return (str.match(/^\s*(define|require)\(/gm) || []).length > 1;
+    }
+
+    /**
+     * Check if needle is a module (path) in define or require statement.
+     * @param {String} needle to search for
+     * @param {String} haystack which might contain needle
+     * @return {Bool}
+     */
+    stringIsPartOfDefineOrRequireStatement(needle, haystack) {
```

Comment:
Yes I agree that's why I tried to add newlines to make it slightly easier to read and of course cover it with unit tests so it can be refactored when needed.